library/std/src/sys/unix/fd.rs

```diff
@@ -425,6 +427,7 @@ impl FileDesc {
         }
     }
     #[cfg(any(target_os = "espidf", target_os = "horizon"))]
+    #[allow(dead_code)]
```

Comment:
Is this still necessary?
Did you figure out why exactly is it reported as dead code now?

Reply:
This is not necessary, it was reported due to misunderstanding with references.